dandiapi/api/views/serializers.py

```diff
@@ -197,3 +197,8 @@ class AssetPathsQueryParameterSerializer(serializers.Serializer):
     path_prefix = serializers.CharField(default='')
     page = serializers.IntegerField(default=1)
     page_size = serializers.IntegerField(default=DandiPagination.page_size)
+
+
+class AssetListSerializer(serializers.Serializer):
+    glob = serializers.CharField(required=False)
+    regex = serializers.CharField(required=False)
```

Comment:
As is, both glob and regex can be specified. We should probably catch this and error in the serializer.

```suggestion
class AssetListSerializer(serializers.Serializer):
    glob = serializers.CharField(required=False)
    regex = serializers.CharField(required=False)
    
    def validate(self, data):
        if 'glob' in data and 'regex' in data:
            raise serializers.ValidationError(
                {'glob': 'Cannot specify both glob and regex'},
            )
        return data
```

Reply:
Why not allow both to be specified?

Reply:
They could conflict and so leaves it ambiguous what should occur. If both are applied, order would matter, and that's fairly opaque to the user. Unless there's a convincing reason to require both be used at the same time I'd suggest we only allow one at a time.

Reply:
Good point, I didn't consider that order would matter - I'll add that validation and another test for this scenario